lib/bundler/cli/add.rb

```diff
@@ -2,34 +2,38 @@
 
 module Bundler
   class CLI::Add
+    attr_reader :gems, :options, :version
+
     def initialize(options, gems)
       @gems = gems
       @options = options
-      @options[:group] = @options[:group].split(",").map(&:strip) if !@options[:group].nil? && !@options[:group].empty?
+      @options[:group] = options[:group].split(",").map(&:strip) if !options[:group].nil? && !options[:group].empty?
+      @version = options[:version].split(",").map(&:strip) unless options[:version].nil?
     end
 
     def run
-      raise InvalidOption, "You can not specify `--strict` and `--optimistic` at the same time." if @options[:strict] && @options[:optimistic]
+      validate_options!
+      dependencies = gems.map {|g| Bundler::Dependency.new(g, version, options) }
 
-      # raise error when no gems are specified
-      raise InvalidOption, "Please specify gems to add." if @gems.empty?
+      Injector.inject(dependencies,
+        :conservative_versioning => options[:version].nil?, # Perform conservative versioning only when version is not specified
+        :optimistic => options[:optimistic],
+        :strict => options[:strict])
 
-      version = @options[:version].nil? ? nil : @options[:version].split(",").map(&:strip)
+      Installer.install(Bundler.root, Bundler.definition) unless options["skip-install"]
+    end
 
-      unless version.nil?
-        version.each do |v|
-          raise InvalidOption, "Invalid gem requirement pattern '#{v}'" unless Gem::Requirement::PATTERN =~ v.to_s
-        end
-      end
+  private
 
-      dependencies = @gems.map {|g| Bundler::Dependency.new(g, version, @options) }
+    def validate_options!
+      raise InvalidOption, "You can not specify `--strict` and `--optimistic` at the same time." if options[:strict] && options[:optimistic]
 
-      Injector.inject(dependencies,
-        :conservative_versioning => @options[:version].nil?, # Perform conservative versioning only when version is not specified
-        :optimistic => @options[:optimistic],
-        :strict => @options[:strict])
+      # raise error when no gems are specified
+      raise InvalidOption, "Please specify gems to add." if gems.empty?
 
-      Installer.install(Bundler.root, Bundler.definition) unless @options["skip-install"]
+      version.to_a.each do |v|
```

Comment:
I removed the if converting version to an array, it's possible to reach here with a nil value.